internal/eval/eval.go

```diff
@@ -3439,3 +3440,30 @@ func evalJSONNUMINCRBY(args []string, store *dstore.Store) []byte {
 	obj.Value = jsonData
 	return clientio.Encode(resultString, false)
 }
+
+func evalTYPE(args []string, store *dstore.Store) []byte {
+	if len(args) != 1 {
+		return diceerrors.NewErrArity("TYPE")
+	}
+	key := args[0]
+	obj := store.Get(key)
+	if obj == nil {
+		return clientio.Encode("none", false)
+	}
+
+	var typeStr string
+	switch obj.TypeEncoding & ObjTypeMask {
```

Comment:
We might be able to use predefined function for extracting the type. This way we can remove `const ObjTypeMask = 0b11110000` as well.
```suggestion
	switch obj.TypeEncoding & ObjTypeMask {
```
```suggestion
        switch oType, _ := object.ExtractTypeEncoding(obj); oType {
```

Reply:
Updated, Really appreciate your suggestion. Thanks.